src/components/RedeemGift/RedeemGift.tsx

```diff
@@ -94,53 +95,57 @@ const RedeemGift = ({ onBackClick, onSuccess }: RedeemGiftProps) => {
   return (
     <WrapperStyled>
       <Header onBackClick={onBackClick} />
-      {isGiftRedeemed ? (
-        <ThankYouPage />
-      ) : (
-        <RedeemGiftWrapperStyled>
-          <SectionHeader center paddingBottom="10px">
-            {t('redeem-gift.button.header', 'Redeem your gift')}
-          </SectionHeader>
-          <InputWrapperStyled>
-            <MyAccountInput
-              error={error}
-              onChange={handleChange}
-              name="giftCode"
-              type="text"
-              value={giftCode}
-              placeholder="XXXX-XXXX"
-            />
-            <Button
-              disabled={!giftCode}
-              theme="confirm"
-              onClickFn={handleVerifyGift}
-            >
-              {isVerifyLoading ? (
-                <Loader buttonLoader color="#ffffff" />
-              ) : (
-                t('redeem-gift.button.verify', 'Verify')
+      {Auth.isLogged() && (
```

Comment:
please avoid nested ternary operator and &&. Use If return instead.

Reply:
refactored, thx for the tip!